### ISSUE TITLE
Gradient in `oklch` space panics

### ISSUE THREAD

**Issue by EpicEricEE** [typst/typst#2902]
### Description

```typ
#set page(width: 15cm, height: auto, margin: 1em)
#set block(width: 100%, height: 1cm, above: 2pt)

// Oklch
#block(fill: gradient.linear(red, purple, space: oklch))
#block(fill: gradient.linear(..color.map.rainbow, space: oklch))
#block(fill: gradient.linear(..color.map.plasma, space: oklch))

#v(1em)

// Oklab
#block(fill: gradient.linear(red, purple, space: oklab))
#block(fill: gradient.linear(..color.map.rainbow, space: oklab))
#block(fill: gradient.linear(..color.map.plasma, space: oklab))
```

PDF export panics with:
```
thread 'main' panicked at crates\typst-pdf\src\color.rs:276:17:
not implemented: Oklch is always converted to Oklab first
stack backtrace:
   0: std::panicking::begin_panic_handler
             at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library\std\src\panicking.rs:595
   1: core::panicking::panic_fmt
             at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library\core\src\panicking.rs:67
   2: typst_pdf::color::impl$1::encode
             at crates\typst-pdf\src\color.rs:276
   3: typst_pdf::color::impl$4::convert<f32>
             at crates\typst-pdf\src\color.rs:436
   4: typst_pdf::gradient::single_gradient
             at crates\typst-pdf\src\gradient.rs:259
   5: typst_pdf::gradient::shading_function
             at crates\typst-pdf\src\gradient.rs:222
   6: typst_pdf::gradient::write_gradients
             at crates\typst-pdf\src\gradient.rs:44
   7: typst_pdf::pdf
             at crates\typst-pdf\src\lib.rs:60
   8: typst::compile::export_pdf
             at crates\typst-cli\src\compile.rs:170
   9: typst::compile::export
             at crates\typst-cli\src\compile.rs:159
  10: typst::compile::compile_once
             at crates\typst-cli\src\compile.rs:103
  11: typst::compile::compile
             at crates\typst-cli\src\compile.rs:61
  12: typst::main
             at crates\typst-cli\src\main.rs:43
  13: core::ops::function::FnOnce::call_once<std::process::ExitCode (*)(),tuple$<> >
             at /rustc/cc66ad468955717ab92600c770da8c1601a4ff33\library\core\src\ops\function.rs:250
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

PNG and SVG is compiled but contains artifacts _(top: oklch, bottom: oklab)_:
![image](https://github.com/typst/typst/assets/7191192/9b031bf5-61c0-4199-979e-2b9eb6aedea8)

This is also visible in the documentation image:
https://typst.app/docs/reference/visualize/gradient/#color-spaces-and-interpolation

### Operating system

Web app, Windows

### Typst version

- [X] I am using the latest version of Typst

**Comment by EpicEricEE** [comment]
PDF panic was introduced with #2730.

Artifacts in PNG and SVG have always been there since oklch has been implemented with #2611. Initially it was only the rainbow, while the grey part was later introduced with #2659.

cc @frozolotl

**Comment by Dherse** [comment]
I'll open up a PR to fix it :)

> EDIT: Turns out the two issues are completely unrelated, as hue interpolation over the 0°/360° boundary is broken for hsv and hsl too.

That shouldn't be the case unless somebody tampered with the special casing 😭 I'll double check it anyway.

**Comment by Dherse** [comment]
Ok, the fix for the panic is in #2913 now I'll look for the color wrapping (that should **not** happen)
